CHANGELOG.md

```diff
@@ -2,15 +2,36 @@
 
 ### Bugfixes
 
-* None.
+* Fix a crash when distinct is applied on two or more properties where
+  the properties contain a link and non-link column.
+  PR [#2979](https://github.com/realm/realm-core/pull/2979).
 
 ### Breaking changes
 
 * None.
 
 ### Enhancements
 
-* None.
+* Parser improvements:
+    - Support for comparing two columns of the same type. For example:
+        - `wins > losses`
+        - `account_balance > purchases.@sum.price`
```

Comment:
@cmelchior hey, isn't this actually [multi-field comparisons](https://github.com/realm/realm-java/issues/1615) and [collection aggregate queries](https://github.com/realm/realm-java/issues/1598) in like one example?

It'd be super-cool if one day, Realm-Java could expose a raw query api that lets you use this stuff 😄

Reply:
That would be super easy. `realm.where(Person.class).filter("wins > losses").findAll()` 😄 . We will probably add this as a complement to the query API relatively quickly. Ideally, I would have liked to add an IntelliJ plugin on top for autocompleting, but that probably has a little lower priority.

Reply:
ooooo that sounds awesome!

The only thing I wonder about is if it is possible to do something crazy like,

``` java
realm.where(Person.class).filter("wins > losses").equalTo("blah", blah").sort("rank").findAll()
```

Reply:
That should be possible. Since the Core part of this just return a Query object on which it should be perfectly possible to continue to work on.